src/hostapi/coreaudio_ios/pa_ios_core.c

```diff
@@ -0,0 +1,1412 @@
+                          void *userData);
+static PaError CloseStream(PaStream * stream);
+static PaError StartStream(PaStream * stream);
+static PaError StopStream(PaStream * stream);
+static PaError AbortStream(PaStream * stream);
+static PaError IsStreamStopped(PaStream * s);
+static PaError IsStreamActive(PaStream * stream);
+static PaTime GetStreamTime(PaStream * stream);
+static OSStatus AudioIOProc(void *inRefCon,
+    AudioUnitRenderActionFlags * ioActionFlags,
+    const AudioTimeStamp * inTimeStamp,
+    UInt32 inBusNumber,
+    UInt32 inNumberFrames,
+    AudioBufferList * ioData);
+static double GetStreamCpuLoad(PaStream * stream);
+
+/*
+ * Callback called when starting or stopping a stream.
+ */
+static void startStopCallback(
+    void *inRefCon,
+    AudioUnit ci,
+    AudioUnitPropertyID inID,
+    AudioUnitScope inScope,
+    AudioUnitElement inElement)
+{
+    PaIosCoreStream *stream = (PaIosCoreStream *) inRefCon;
+    UInt32 isRunning;
+    UInt32 size = sizeof(isRunning);
+    OSStatus err;
+
+    err = AudioUnitGetProperty(ci, kAudioOutputUnitProperty_IsRunning, inScope, inElement, &isRunning, &size);
+    assert(!err);
+    if (err)
+        isRunning = false;
+    if (isRunning)
+        return;
+    if (stream->inputUnit && stream->outputUnit && stream->inputUnit != stream->outputUnit && ci == stream->inputUnit)
+        return;
+    PaStreamFinishedCallback *sfc = stream->streamRepresentation.streamFinishedCallback;
+
+    if (stream->state == STOPPING)
+        stream->state = STOPPED;
+    if (sfc)
+        sfc(stream->streamRepresentation.userData);
+}
+
+static void FillDeviceInfo(
+    PaIosAUHAL * auhalHostApi,
+    PaDeviceInfo * deviceInfo,
+    PaHostApiIndex hostApiIndex)
+{
+    memset(deviceInfo, 0, sizeof(PaDeviceInfo));
+
+    deviceInfo->structVersion = 2;
+    deviceInfo->hostApi = hostApiIndex;
+    deviceInfo->name = "Default";
+    deviceInfo->defaultSampleRate = 48000;
+    deviceInfo->maxInputChannels = 1;
+    deviceInfo->maxOutputChannels = 2;
+
+    deviceInfo->defaultLowInputLatency = 0.008;
+    deviceInfo->defaultHighInputLatency = 0.080;
+    deviceInfo->defaultLowOutputLatency = 0.008;
+    deviceInfo->defaultHighOutputLatency = 0.080;
+}
+
+PaError PaIosCore_Initialize(PaUtilHostApiRepresentation ** hostApi, PaHostApiIndex hostApiIndex)
+{
+    PaError result = paNoError;
+    PaIosAUHAL *auhalHostApi = NULL;
+    PaDeviceInfo *deviceInfoArray;
+
+    auhalHostApi = (PaIosAUHAL *) PaUtil_AllocateZeroInitializedMemory(sizeof(PaIosAUHAL));
+    if (auhalHostApi == NULL) {
+        result = paInsufficientMemory;
+        goto error;
+    }
+    auhalHostApi->allocations = PaUtil_CreateAllocationGroup();
+    if (auhalHostApi->allocations == NULL) {
+        result = paInsufficientMemory;
+        goto error;
+    }
+    *hostApi = &auhalHostApi->inheritedHostApiRep;
+
+    (*hostApi)->info.structVersion = 1;
+    (*hostApi)->info.type = paCoreAudio;
```

Comment:
Should iOS be using `paInDevelopment` and later get its own type id or is it okay to use the same ID as the macOS Core Audio implementation, given that the two should never be available at the same time?